src/main/scala/lantern/TensorDifferentiation.scala

```diff
@@ -2818,57 +2817,75 @@ trait TensorDslCublas extends TensorDsl with GPUOps {
       Tensor(res, m, n)
     }
 
-    def launchBinaryKernel(n: Int, x: Tensor, y: Tensor, res: Rep[Array[Float]])
-                          (op: (String, String) => String): Unit = {
+    def launchBinaryKernel(x: Tensor, y: Tensor)(op: (String, String) => String): Tensor = {
```

Comment:
`op` should be reformulated as a staged function of type `Rep[(Float, Float) => Float]`.